CONTRIBUTING.md

```diff
@@ -1,49 +1,122 @@
-|:-----------------|:-----------|:-------------------|:---------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------|
-| `main`           | YES        | N/A                | What is live in production (**stable**).<br/>A pull request is required to merge code into `main`.                                                                                                 |
-| `dev`            | YES        | `main`             | The latest state of development (**unstable**).<br/>A pull request is required to merge code into `dev`                                                                                            |
-| `feature/*`      | NO         | `dev`              | Cutting-edge features (**unstable**). These branches are used for any maintenance features / active development.                                                                                   |
-| `release/vX.Y.Z` | NO         | `dev`              | A temporary release branch that follows the [semver](http://semver.org/) versioning. This is what is sent to UAT.<br/>A pull request is required to merge code into any `release/vX.Y.Z` branch.   |
-| `hotfix/*`       | NO         | `main`             | These are bug fixes against production.<br/>This is used because develop might have moved on from the last published state.<br/>Remember to merge this back into develop and any release branches. |
+### 1. Creating Feature #1 (Developer #1)
 
+#### Step 1: Clone the Repository
+Clone the repository to your local machine.
+```bash
+git clone git@github.com:aws-solutions-library-samples/osml-imagery-toolkit.git
+```
+
+#### Step 2: Create a New Branch
+Create a new branch off `main`, naming it related to the work being done.
+```bash
+git checkout -b feature/<feature_id>
+```
+
+#### Step 3: Develop and Commit Changes
+On the new branch, code and add commits as necessary.
+
+#### Step 4: Push the New Branch
+Push the new branch to trigger unit tests, static code analysis, Sonar cube checks, security checks, etc.,
+before merging to `main`.
+```bash
+git push origin feature/<feature_id>
+```
+
+#### Step 5: Create a Pull Request (PR)
+Open a PR against `main` to kick off a discussion.
+Open a PR as soon as possible, even if it's not ready (mark it as WIP).
+This ensures ample time for review and discussion, improving code quality.
+
+#### Step 6: PR Review and Merge
+After your PR is reviewed, make any necessary changes by repeating steps 2 and 3.
+A team member will merge your PR after approval,
+ensuring it's reviewed by someone who has not contributed to the branch.
+
+### 2. Creating Feature #2 (Developer #2) and Rebasing
```

Comment:
Do we really need this two developer scenario. Why don't we just assume that something happened to main while the first developer was working on his feature and show them rebasing their feature branch right before creating the PR?

Reply:
I can get rid of the developer 2 scenario - it was probably a bit overkill but I was trying to be explicit.